### PR TITLE
Packaged and updates charts. 

### PR DESCRIPTION
**Pull request by sfxworks**
Resolves https://github.com/kubernetes-sigs/federation-v2/issues/701

Currently uses repo as a base. Consider using Github pages or an existing bucket to host the chart? If so, before the merge, the chart would need to be updated with the new endpoint.

### CODE REVIEW

File: charts/federation-v2/README.md [hunk 1]
```diff
@@ -29,24 +29,37 @@ group](https://groups.google.com/forum/#!forum/kubernetes-sig-multicluster).
 
 ## Installing the Chart
 
-First you'll need to create the reserved namespace for registering clusters with the
-cluster registry:
+First you'll need to create the reserved namespace for registering clusters with the cluster registry:
```
Comment (marun) [inline charts/federation-v2/README.md]: This is an unrelated formatting change.

File: charts/federation-v2/README.md [hunk 2]
```diff
@@ -29,24 +29,37 @@ group](https://groups.google.com/forum/#!forum/kubernetes-sig-multicluster).
...
 
 ```bash
 $ kubectl create ns kube-multicluster-public
 ```
 
-Install the chart with the release name `federation-v2`:
+Next, add the chart to your local resitory.
```
Comment (marun) [inline charts/federation-v2/README.md]: s/resitory/repository/

File: charts/federation-v2/README.md [hunk 3]
```diff
@@ -29,24 +29,37 @@ group](https://groups.google.com/forum/#!forum/kubernetes-sig-multicluster).
...
+```bash
+$ helm repo add federation-v2 https://raw.githubusercontent.com/kubernetes-sigs/federation-v2/master/charts/federation-v2
+
+$ helm repo list
+NAME            URL                                        
+federation-v2   https://raw.githubusercontent.com/kubernetes-sigs/federation-v2/master/charts/federation-v2
+```
+
+With the chart installed, availble versions can be viewed.
```
Comment (marun) [inline charts/federation-v2/README.md]: Did you mean 'With the repo added' ?

Also, s/availble/available/

File: charts/federation-v2/README.md [hunk 4]
```diff
@@ -29,24 +29,37 @@ group](https://groups.google.com/forum/#!forum/kubernetes-sig-multicluster).
...
+```bash
+$ helm search federation-v2
+```
 
+Install the chart and append the version you would like using the `--version` tag. Here, we specify version 0.0.7 and name it federation-v2.
+Note, the chart will default to 0.0.7.
```
Comment (marun) [inline charts/federation-v2/README.md]: The chart should not default to 0.0.7.  If there is a default, it should be to the latest released version.

File: charts/federation-v2/README.md [hunk 5]
```diff
@@ -29,24 +29,37 @@ group](https://groups.google.com/forum/#!forum/kubernetes-sig-multicluster).
...
+First you'll need to create the reserved namespace for registering clusters with the cluster registry:
 
 ```bash
 $ kubectl create ns kube-multicluster-public
 ```
 
-Install the chart with the release name `federation-v2`:
+Next, add the chart to your local resitory.
+```bash
+$ helm repo add federation-v2 https://raw.githubusercontent.com/kubernetes-sigs/federation-v2/master/charts/federation-v2
```
Comment (marun) [inline charts/federation-v2/README.md]: Would it make sense to suggest adding it as `fedv2-repo` or something that differentiates from the name of the chart?

File: charts/federation-v2/Chart.yaml [hunk 6]
```diff
@@ -1,5 +1,5 @@
 apiVersion: v1
-appVersion: 0.0.2
+appVersion: 0.0.7
```
Comment (marun) [inline charts/federation-v2/Chart.yaml]: This entire commit needs to be removed.  

 - the chart version should not be updated in the tree.  It should only be set in the archived chart.
 - checking large binary files into git is a no-no.  It inflates the size of the tree unnecessarily.
 - The index will need to be created as part of the release process.

File: charts/federation-v2/values.yaml [hunk 7]
```diff
@@ -10,7 +10,7 @@ controllermanager:
   replicaCount: 1
   repository: quay.io/kubernetes-multicluster
   image: federation-v2
-  tag: canary
+  tag: v0.0.7
```
Comment (marun) [inline charts/federation-v2/values.yaml]: Please remove this commit - the image tag should remain canary in master.  Only the archived chart should have its tag set, and the tag should reflect the release the chart is intended to be a part of.

File: docs/releasing.md [hunk 8]
```diff
@@ -24,6 +24,13 @@ Creating a federation v2 release involves the following steps:
...
+   1.  Adjust the default image tag in values.yaml
+   2.  Update the chart version in Chart.yaml
+   3.  Package the chart using `helm package federation-v2`
+   4.  Place the release packaged chart (.tgz file) in the federation-v2/ folder.
+   5.  Add the new version to the chart index `helm repo index federation-v2/ --url=https://raw.githubusercontent.com/kubernetes-sigs/federation-v2/master/charts/federation-v2/`
+   6.  Ensure index.yaml containes the latest release within the entries array.
 4. Create github release
    - Copy text from old release and replace old tag references
-   - Add `kubefed2.tar.gz` and `kubefed2.tar.gz.sha`
+   - Add `kubefed2.tar.gz` and `kubefed2.tar.gz.sha`
```
Comment (marun) [inline docs/releasing.md]: This appears to be an unrelated formatting change.  Also, an additional line instructing the inclusion of `federation-v2-chart.tar.gz` and `federation-v2-chart.tar.gz.sha` should be added.

File: docs/releasing.md [hunk 9]
```diff
@@ -24,6 +24,13 @@ Creating a federation v2 release involves the following steps:
    3. `kubefed2 version` (check that the output is as expected)
    4. `tar cvzf kubefed2.tar.gz kubefed2`
    5. `sha256sum kubefed2.tar.gz > kubefed2.tar.gz.sha`
+4. Package the helm chart for release
+   1.  Adjust the default image tag in values.yaml
+   2.  Update the chart version in Chart.yaml
+   3.  Package the chart using `helm package federation-v2`
```
Comment (marun) [inline docs/releasing.md]: As per the kubefed2 instructions, please specify the command to execute and then explain it in brackets, e.g

`helm package federation-v2` (create a tar archive of the chart)

File: docs/releasing.md [hunk 10]
```diff
@@ -24,6 +24,13 @@ Creating a federation v2 release involves the following steps:
...
+   4.  Place the release packaged chart (.tgz file) in the federation-v2/ folder.
```
Comment (marun) [inline docs/releasing.md]: This line should be replaced with

`sha256sum federation-v2-chart.tar.gz > federation-v2-chart.tar.gz.sha`

File: docs/releasing.md [hunk 11]
```diff
@@ -24,6 +24,13 @@ Creating a federation v2 release involves the following steps:
...
+   5.  Add the new version to the chart index `helm repo index federation-v2/ --url=https://raw.githubusercontent.com/kubernetes-sigs/federation-v2/master/charts/federation-v2/`
+   6.  Ensure index.yaml containes the latest release within the entries array.
```
Comment (marun) [inline docs/releasing.md]: Once the version has been added to the chart index, it will be necessary to revert the changes from steps 1 and 2 (they should never be committed to master) and create a PR that captures the changes to index.yaml.

File: docs/releasing.md [hunk 12]
```diff
@@ -24,6 +24,13 @@ Creating a federation v2 release involves the following steps:
    3. `kubefed2 version` (check that the output is as expected)
    4. `tar cvzf kubefed2.tar.gz kubefed2`
    5. `sha256sum kubefed2.tar.gz > kubefed2.tar.gz.sha`
+4. Package the helm chart for release
```
Comment (marun) [inline docs/releasing.md]: Please update the instructions for kubefed to use `.tgz` instead of `.tar.gz` for parity with the chart archive name.

**Comment by marun** [comment]
@sfxworks I've released [v0.0.8](https://github.com/kubernetes-sigs/federation-v2/releases/tag/v0.0.8) and proposed a [corresponding PR](https://github.com/kubernetes-sigs/federation-v2/pull/754) that adds the repo index.  Please update this PR to reflect the new location of the repo index (`charts` instead of `charts/federation-v2`).

File: charts/federation-v2/README.md (hunk 5 above)
Comment (sfxworks) [reply charts/federation-v2/README.md]: Yes, and it would open room for any new charts this repo would have in the future. Working with #754 seems to work perfectly. 

https://raw.githubusercontent.com/kubernetes-sigs/federation-v2/add-chart-repo-index/charts/

This path assumes a new branch named "add-chart-repo-index" is going to be made. Can you confirm this?

File: charts/federation-v2/Chart.yaml (hunk 6 above)
Comment (sfxworks) [reply charts/federation-v2/Chart.yaml]: Removing.

File: charts/federation-v2/values.yaml (hunk 7 above)
Comment (sfxworks) [reply charts/federation-v2/values.yaml]: Removing.

**Comment by sfxworks** [comment]
Updated PR based on comments.

File: docs/releasing.md (hunk 12 above)
Comment (sfxworks) [reply docs/releasing.md]: Updated.

File: docs/releasing.md (hunk 11 above)
Comment (sfxworks) [reply docs/releasing.md]: Adjusted. `   7.  Create PR that captures index.yaml for branch "add-chart-repo-index"` Acceptable?

File: docs/releasing.md (hunk 10 above)
Comment (sfxworks) [reply docs/releasing.md]: Updated `   4.  `sha256sum federation-v2-<x.x.x>.tgz > federation-v2-<x.x.x>.tgz` (Name and checksum the packaged chart accordingly)`

File: docs/releasing.md (hunk 9 above)
Comment (sfxworks) [reply docs/releasing.md]: Adjusted all instructions

File: docs/releasing.md (hunk 8 above)
Comment (sfxworks) [reply docs/releasing.md]: Unknown how that occurred...
Added: `   - Add `federation-v2-<x.x.x>.tgz` and `federation-v2-<x.x.x>.tgz.sha`

File: charts/federation-v2/README.md (hunk 4 above)
Comment (sfxworks) [reply charts/federation-v2/README.md]: Removed default note.

File: charts/federation-v2/README.md (hunk 3 above)
Comment (sfxworks) [reply charts/federation-v2/README.md]: Yes! Thank you for that catch.

**Comment by sfxworks** [comment]
Reviewed and reupdated PR based on comments.

**Comment by marun** [comment]
@sfxworks Thank you for responding to my comments.  Please squash to a single commit and add back `charts/federation-v2/Chart.yaml`.

**Comment by sfxworks** [comment]
Readded and squashed! 👍

**Comment by sfxworks** [comment]
Readded and squashed 👍

File: charts/federation-v2/README.md [hunk 13]
```diff
@@ -36,17 +36,30 @@ cluster registry:
 $ kubectl create ns kube-multicluster-public
 ```
 
-Install the chart with the release name `federation-v2`:
+Next, add the chart to your local repository.
+```bash
+$ helm repo add federation-v2 https://raw.githubusercontent.com/kubernetes-sigs/federation-v2/add-chart-repo-index/charts/
```
Comment (marun) [inline charts/federation-v2/README.md]: - From subsequent commands the repo name should be `fedv2-charts` instead of `federation-v2`.
- The url should be `https://raw.githubusercontent.com/kubernetes-sigs/federation-v2/master/charts`

File: charts/federation-v2/README.md [hunk 14]
```diff
@@ -36,17 +36,30 @@ cluster registry:
...
+
+$ helm repo list
+NAME            URL                                        
+federation-v2   https://raw.githubusercontent.com/kubernetes-sigs/federation-v2/add-chart-repo-index/charts/
```
Comment (marun) [inline charts/federation-v2/README.md]: ditto the previous comment on repo name and url

File: docs/releasing.md [hunk 15]
```diff
@@ -22,8 +22,17 @@ Creating a federation v2 release involves the following steps:
...
+   4. `tar cvzf kubefed2.tgz kubefed2`
+   5. `sha256sum kubefed2.tgz > kubefed2.tgz.sha`
+4. Package the helm chart for release
+   1.  Adjust the default image tag in values.yaml
+   2.  Update the chart version in Chart.yaml
+   3.  `helm package federation-v2` (Package the chart)
+   4.  `sha256sum federation-v2-<x.x.x>.tgz > federation-v2-<x.x.x>.tgz` (Name and checksum the packaged chart accordingly)
+   5.  `helm repo index federation-v2/ --url=https://raw.githubusercontent.com/kubernetes-sigs/federation-v2/add-chart-repo-index/charts/` (Add the new version to the chart index)
+   6.  Ensure index.yaml containes the latest release within the entries array
+   7.  Create PR that captures index.yaml for branch "add-chart-repo-index"
```
Comment (marun) [inline docs/releasing.md]: `s/Create PR.*/Propose a PR that updates index.yaml/`

File: docs/releasing.md [hunk 16]
```diff
@@ -22,8 +22,17 @@ Creating a federation v2 release involves the following steps:
...
-   4. `tar cvzf kubefed2.tar.gz kubefed2`
-   5. `sha256sum kubefed2.tar.gz > kubefed2.tar.gz.sha`
+   4. `tar cvzf kubefed2.tgz kubefed2`
+   5. `sha256sum kubefed2.tgz > kubefed2.tgz.sha`
+4. Package the helm chart for release
+   1.  Adjust the default image tag in values.yaml
+   2.  Update the chart version in Chart.yaml
+   3.  `helm package federation-v2` (Package the chart)
+   4.  `sha256sum federation-v2-<x.x.x>.tgz > federation-v2-<x.x.x>.tgz` (Name and checksum the packaged chart accordingly)
+   5.  `helm repo index federation-v2/ --url=https://raw.githubusercontent.com/kubernetes-sigs/federation-v2/add-chart-repo-index/charts/` (Add the new version to the chart index)
```
Comment (marun) [inline docs/releasing.md]: The command should be: `helm repo index .  --merge index.yaml --url=https://github.com/kubernetes-sigs/federation-v2/releases/download/v<x.x.x>/federation-v2-<x.x.x>.tgz`

File: docs/releasing.md [hunk 17]
```diff
@@ -22,8 +22,17 @@ Creating a federation v2 release involves the following steps:
...
    2. `cd bin`
    3. `kubefed2 version` (check that the output is as expected)
-   4. `tar cvzf kubefed2.tar.gz kubefed2`
-   5. `sha256sum kubefed2.tar.gz > kubefed2.tar.gz.sha`
+   4. `tar cvzf kubefed2.tgz kubefed2`
+   5. `sha256sum kubefed2.tgz > kubefed2.tgz.sha`
+4. Package the helm chart for release
+   1.  Adjust the default image tag in values.yaml
+   2.  Update the chart version in Chart.yaml
+   3.  `helm package federation-v2` (Package the chart)
```
Comment (marun) [inline docs/releasing.md]: step 3 should be 'cd charts`

**Comment by marun** [comment]
Closing in favor of #793
